### PR TITLE
Improve relocation record checks inside genLoadLongConstant

### PR DESCRIPTION
**Pull request by dchopra001**
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

**Comment by dchopra001** [comment]
Some of these checks may be too aggressive. I'd like to test those out and make sure they are absolutely necessary before this PR will be ready for a review.

**Comment by dchopra001** [comment]
@fjeremic @mpirvu FYI This is ready for review.

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -234,7 +234,7 @@ genLoadLongConstant(TR::CodeGenerator * cg, TR::Node * node, int64_t value, TR::
    //disable LARL for AOT for easier relocation
    //cannot safely generate LARL for longs on 31-bit
    else if (cg->canUseRelativeLongInstructions(value) &&
-            comp->target().is64Bit() && !(comp->compileRelocatableCode()))
+            comp->target().is64Bit() && !(comp->compileRelocatableCode() || comp->isOutOfProcessCompilation()))
```
Comment (dchopra001) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:237]: The `comp->isOutOfProcessCompilation()` here will need to be replaced with something more appropriate.

**Comment by dchopra001** [comment]
Note: I have verified these tests with internal builds w.r.t JITServer testing. Except for an existing issue, the builds are stable for: `_sanity.functional.regular, _sanity.functional.native,` and also for cold/warm runs of AcmeAir.

File: compiler/z/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (fjeremic) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:237]: @dchopra001 you're right, that is the correct solution. I would override `canUseRelativeLongInstructions` in OpenJ9 and return `false` if we are a remote compile.

Comment (dchopra001) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:237]: It seems like this `isOutOfProcessCompilation()` now exists in OMR:
https://github.com/eclipse/omr/blob/ff6b4b2f91eab5054d0644d76a20f98ba168b4c6/compiler/compile/OMRCompilation.hpp#L1054-L1060

So we have the option of leaving this code as is and things will still work.  We can still override `canUseRelativeLongInstructions` if you prefer that approach. Let me know.

Comment (fjeremic) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:237]: This is not the only place that uses relative long instructions, i.e. there are other locations querying `canUseRelativeLongInstructions`, so my suggestion is to override this API in OpenJ9 and return false if we are a remote compile.

Comment (dchopra001) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: Check removed here: aabd9acc9738af958cff3ae5ec9ea9e6206ddb8e.
Related OpenJ9 PR: https://github.com/eclipse/openj9/pull/10113

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by AdamBrousseau** [comment]
genie-omr build all

**Comment by AdamBrousseau** [comment]
genie-omr build all

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**